crates/release_plz_core/src/command/release_pr/mod.rs

```diff
@@ -456,25 +458,33 @@ async fn github_force_push(
 /// Temporary branch.
 /// It deletes the branch in remote when it goes out of scope.
 /// In this way, we can ensure that the branch is deleted even if the program panics.
-struct TmpBranch<'a> {
+struct TmpBranch {
     name: String,
-    repository: &'a Repo,
+    client: Arc<GitClient>,
 }
 
-impl<'a> TmpBranch<'a> {
-    fn checkout_new(repository: &'a Repo, name: impl Into<String>) -> anyhow::Result<Self> {
+impl TmpBranch {
+    fn checkout_new(
+        repository: &Repo,
+        name: impl Into<String>,
+        client: Arc<GitClient>,
+    ) -> anyhow::Result<Self> {
         let name = name.into();
         repository.checkout_new_branch(&name)?;
-        let branch = Self { name, repository };
+        let branch = Self { name, client };
         Ok(branch)
     }
 }
 
-impl Drop for TmpBranch<'_> {
+impl Drop for TmpBranch {
     fn drop(&mut self) {
-        if let Err(e) = self.repository.delete_branch_in_remote(&self.name) {
-            tracing::error!("cannot delete branch {}: {:?}", self.name, e);
-        }
+        let client = self.client.clone();
+        let name = self.name.clone();
+        tokio::spawn(async move {
+            if let Err(e) = client.delete_branch(&name).await {
+                tracing::error!("cannot delete branch {name}: {e:?}");
+            }
+        });
```

Comment:
```suggestion
        // Create a new runtime to block on the async operation
        match tokio::runtime::Runtime::new() {
            Ok(rt) => {
                rt.block_on(async move {
                    if let Err(e) = client.delete_branch(&name).await {
                        tracing::error!("cannot delete branch {name}: {e:?}");
                    }
                });
            }
            Err(e) => {
                tracing::error!("failed to create runtime for branch deletion: {e:?}");
                return;
            }
        }
```
what do you think about doing this to wait for the completion?

Reply:
mmmh, why do you say it waits?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=0943eec06529e6fcf467fbfd139edac1
<img width="3224" height="2400" alt="Image" src="https://github.com/user-attachments/assets/719899dd-1cbd-4f57-bc62-3a095316e26d" />

Reply:
Oh sorry, I meant the previous implementation with Git CLI.

Reply:
ok, what do you think about block_on vs spawn?
With block_on we wait so we make sure that the branch is deleted, right? 🤔
With spawn it might happen that release-plz exits because main finishes, but we don't log a possible error for example.

Reply:
I checked the `block_on` implementation, but it panics.

```
thread 'main' panicked at /playground/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.47.1/src/runtime/scheduler/multi_thread/mod.rs:86:9:
Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.
```
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=b329a7104b937a9c3b090566af69b822

The `spawn` one works fine, but yeah, it might not log errors if the main thread exits earlier.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=973c19f768259c71a13a10e26569e0a7

Reply:
If I remove the `TmpBranch`, it would be like this.

```rust
async fn github_force_push(
    client: Arc<GitClient>,
    pr: &GitPr,
    repository: &Repo,
) -> anyhow::Result<()> {
    // Create a temporary branch.
    let tmp_release_branch = format!("{}-tmp-{}", pr.branch(), rand::random::<u32>());
    repository.checkout_new_branch(&tmp_release_branch)?;

    let run = async {
        // Push the "Verified" commit in the temporary branch using
        // the GitHub API.
        // We push the release-plz changes to the temporary branch instead of the release PR branch because:
        // - You can't force-push with the GitHub API, so we can't commit to the release PR branch
        //   directly if we want a "Verified" commit.
        // - If we revert the last commit of the release PR branch, GitHub will close the release PR
        //   because the branch is the same as the default branch. So we can't revert the latest release-plz commit and push the new one.
        // To learn more, see https://github.com/release-plz/release-plz/issues/1487
        github_create_release_branch(&client, repository, &tmp_release_branch, &pr.title).await?;

        repository.fetch(&tmp_release_branch)?;

        // Rewrite the PR branch so that it's the same as the temporary branch.
        repository.force_push(&format!(
            "{}/{}:{}",
            repository.original_remote(),
            tmp_release_branch,
            pr.branch()
        ))?;
        Ok::<(), anyhow::Error>(())
    };

    let result = run.await;

    if let Err(e) = client.delete_branch(&tmp_release_branch).await {
        tracing::error!("cannot delete branch {tmp_release_branch}: {e:?}");
    }

    result
}
```

Reply:
sure, let's remove TmpBranch. Please extract the "run" async block to a function because it looks a bit weird imo

Reply:
maybe this would allow to remove the Arc?

Reply:
Refactored! I didn't have a good idea for the function name